File: D2.UserManagement/ServiceRegistration.fs
namespace D2.UserManagement

open D2.Common
open Microsoft.Extensions.Logging
open Newtonsoft.Json
open System

module ServiceRegistration =

    type EndPoint() =
        member val Name = String.Empty with get, set
        member val Uri = String.Empty with get, set

    type Service() =
        member val Name = "UserManagement" with get
        member val BaseUrl = (
                              ServiceConfiguration.configuration.Hosting
                              |> Seq.head
                             ).FullAddress
                             
        member val Version = ServiceConfiguration.versionInfo.Version with get
        member val Patch = ServiceConfiguration.versionInfo.Patch with get
        member val EndPoints = [|
                                   EndPoint(Name = "Frontend", Uri = "/");
                                   EndPoint(Name = "Register", Uri = "/users/register");
                               |] with get

    let registerSelf (logger : ILogger) =
        let textData = JsonConvert.SerializeObject(Service ())

        ServiceRegistrator.registerSelf logger textData

File: D2.MasterData/ServiceRegistration.cs
using D2.Common;
using Microsoft.Extensions.Logging;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace D2.MasterData
{
    public static class ServiceRegistration
    {
        public class EndPoint
        {
            public string Name { get; set; }
            public string Uri { get; set; }
        }

        public class Service
        {
            public string Name => "administrationunits";
            public string BaseUrl => ServiceConfiguration.configuration.Hosting.First().FullAddress;
            public string Group => "md";
            public int Version => ServiceConfiguration.versionInfo.Version;
            public int Patch => ServiceConfiguration.versionInfo.Patch;
            public EndPoint[] EndPoints => new[] {
                                   new EndPoint { Name = "list_administrationunits", Uri = "/AdministrationUnit/list" },
                                   new EndPoint { Name = "create_administrationunit", Uri = "/AdministrationUnit/create" }
                              };
        }

        public static bool registerSelf(ILogger logger)
        {
            var textData = JsonConvert.SerializeObject(new Service());

            return ServiceRegistrator.registerSelf(logger, textData);
        }
    }
}
